docs/markdown/snippets/dir_options_outside_prefix.md

```diff
@@ -0,0 +1,9 @@
+It is now allowed to set directory options to paths outside of prefix.
+This bring us in line with Autotools and CMake and it is useful for
+platforms like Nix, which install projects into multiple independent prefixes.
+
+As a consequence, `get_option` might return absolute paths for some
+directory options, if a directory outside of prefix is passed. This
+is technically a backwards incompatible change but its effect
```

Comment:
Arguably it is not backwards-incompatible, because it depends on the user configuration, right? And so Meson actually does the same thing in all cases, with the same return values of `get_option()`, ***except*** in cases where currently it would report an error and refuse to continue.

Reply:
Yes, I meant incompatibility in a sense that previously correct Meson build scripts may become incorrect because of the changed assumptions.